### PR TITLE
Python 3.7 upgrade

### PR DESCRIPTION
**Pull request by aclegg3**
## Motivation and Context

Python 3.6 is past end of life, time to upgrade.

## How Has This Been Tested

CI

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 [![Conda Platforms support Badge](https://img.shields.io/conda/pn/aihabitat/habitat-sim?color=orange&label=platforms)](https://anaconda.org/aihabitat/habitat-sim)
 [![Documentation](https://img.shields.io/badge/docs-automated-green.svg)](https://aihabitat.org/docs/habitat-sim/)
 [![pre-commit](https://img.shields.io/badge/pre--commit-enabled-brightgreen?logo=pre-commit&logoColor=white)](https://github.com/pre-commit/pre-commit)
-[![Python 3.6, 3.7, 3.8](https://img.shields.io/badge/python-3.6%20%7C%203.7%20%7C%203.8-blue.svg)](https://www.python.org/downloads/release/)
+[![Python 3.7, 3.8](https://img.shields.io/badge/python-3.6%20%7C%203.7%20%7C%203.8-blue.svg)](https://www.python.org/downloads/release/)
```
Comment (Skylion007) [inline README.md:8]: We aren't enabling 3.9? Should work out of the box.

File: conda-build/README.md [hunk 2]
```diff
@@ -1,9 +1,9 @@
 ### Building for macOS
 
-Install conda packages as decribed here ```conda-build/common/install_conda.sh``` and run ```conda create --name py36 python=3.6 -y;conda activate py36```.
-Running ```python macos_matrix_builder.py``` with python >= 3.6 will start the build process by setting all the environment variables and making a call to conda build. Make sure the meta.yaml file in conda-build/habitat-sim/ is configured correctly accoridng to https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html
+Install conda packages as decribed here ```conda-build/common/install_conda.sh``` and run ```conda create --name py36 python=3.7 -y;conda activate py36```.
```
Comment (Skylion007) [inline conda-build/README.md:3]:
```suggestion
Install conda packages as decribed here ```conda-build/common/install_conda.sh``` and run ```conda create --name py37 python=3.7 -y;conda activate py37```.
```

File: setup.py [hunk 3]
```diff
@@ -463,7 +463,7 @@ def load(filename):
         package_dir={"": "src_python"},
         install_requires=requirements,
         tests_require=["hypothesis", "pytest-benchmark", "pytest"],
-        python_requires=">=3.6",
+        python_requires=">=7",
```
Comment (Skylion007) [inline setup.py:466]:
```suggestion
        python_requires=">=3.7",
```

**Review by Skylion007 — COMMENTED** [review]
There is one nice change we can finally do if we upgrade 3.7 and this is to de-stringify circular dependencies in MyPy type with `from future import annotations.` I will add to a pre-commit check to the PR to enforce that.

**Review by jturner65 — APPROVED** [review]
Hooray for this! LGTM once everything passes.

**Comment by aclegg3** [comment]
Thanks for the additional fixes, @Skylion007.

**Pull request merged**